caproto/_log.py

```diff
@@ -257,14 +222,16 @@ def filter(self, record):
 
 
 class RoleFilter(logging.Filter):
-    def __init__(self, roles):
-        self.roles = roles
+    def __init__(self, role, level='NOTSET', exclusive=False):
+        self.roles = role
+        self.levelno = logging.getLevelName(level)
```

Comment:
It should be allowed for the user to pass in `level=10`. Can you put this conversion inside an `if` block that checks if the input is a string?